detection_rules/rule.py

```diff
@@ -768,6 +768,23 @@ class HistoryWindowStart:
     type: Literal["new_terms"]
     new_terms: NewTermsMapping
 
+    @pre_load
+    def preload_data(self, data: dict, **kwargs) -> dict:
+        """Preloads and formats the data to match the required schema."""
+        if "new_terms_fields" in data and "history_window_start" in data:
+            new_terms_mapping = {
+                "field": "new_terms_fields",
+                "value": data.pop("new_terms_fields"),
+                "history_window_start": [
+                    {
+                        "field": "history_window_start",
+                        "value": data.pop("history_window_start")
+                    }
+                ]
+            }
```

Comment:
a nitpick: I would separate object creation from `data` mutation -- it's easy to overlook `pop()` calls if it's all in one.

Keeping it super explicit:
```python
            new_terms_mapping = {
                "field": "new_terms_fields",
                "value": data["new_terms_fields"],
                "history_window_start": [
                    {
                        "field": "history_window_start",
                        "value": data["history_window_start"]
                    }
                ]
            }
            # A note while these fields need to be dropped
            data.pop("new_terms_fields")
            data.pop("history_window_start")
```
            
```